### PR TITLE
Add mvp flag and redirect to eforms for out of scope journeys

### PR DESCRIPTION
**Pull request by oliverweighell**

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -12,6 +12,14 @@ const offencesList = require('./data/offence_list');
 const https = require('https');
 const utils = require('./utils');
 
+router.use((req, res, next) => {
+  let mvpFlag = req.query && req.query.mvp;
```
Comment (arthurashman) [inline app/routes.js:16]: Something Ruby-y that looks like it's been added to js now - Optional Chaining - I think would work and be more consise here
`req?.query?.mvp`

Comment (arthurashman) [reply app/routes.js:16]: https://stackoverflow.com/questions/32139078/null-safe-property-access-and-conditional-assignment-in-es6-2015

File: app/views/index.html [hunk 2]
```diff
@@ -18,10 +18,10 @@ <h1 class="govuk-heading-xl">
       </p>
       <ul class="govuk-list govuk-list--bullet">
         <li>
-          <a href="/laa_portal">Start an application (MVP)</a>
+          <a href="/laa_portal?mvp=true">Start an application (MVP)</a>
```
Comment (arthurashman) [inline app/views/index.html:21]: Thoughts on removing (MVP) and (beyond MVP) brackets if we've got the mvp param?

File: app/routes.js [hunk 3]
```diff
@@ -88,12 +98,33 @@ router.post('/dwp_passported', function (req, res, next) {
...
   _.set(req.session.data, 'means_assessment.benefits_status.passported', true);
-  res.render('dwp_passported');
+
+  let mvp = req.session.mvp;
+  let outOfScope = false;
+  if (mvp) {
+    let nino = _.get(req.session.data, 'client_details.client.national_insurance_number');
+    let partner = _.get(req.session.data, 'client_details.partner.first_name');
+    let hasPartner = !_.isEmpty(partner);
+    if (_.isEmpty(nino) || hasPartner) {
```
Comment (arthurashman) [inline app/routes.js:108]: I'm assuming `if (_.isEmpty(nino) || _.isEmpty(partner))` didn't work?

Comment (arthurashman) [reply app/routes.js:108]: All the double negatives got a bit confusing is all

**Review by arthurashman — COMMENTED** [review]
Only added some thoughts, logic-wise LGTM - I'd be keen for @carolinapizatto to confirm this is right though

File: app/routes.js (hunk 3 above)
Comment (oliverweighell) [reply app/routes.js:108]: The condition is checking if partner is NOT empty (ie if they have a partner, they slip out of MVP scope). I could use !_.isEmpty(partner) but my thinking was that `hasPartner` would be more explicit and easier to read.

**Pull request merged**